samples/kvssink_gstreamer_sample.cpp

```diff
@@ -715,12 +792,16 @@ int gstreamer_init(int argc, char *argv[], CustomData *data) {
     g_signal_connect(G_OBJECT(bus), "message::error", (GCallback) error_cb, data);
     g_signal_connect(G_OBJECT(bus), "message::eos", G_CALLBACK(eos_cb), data);
     gst_object_unref(bus);
+
+    // Create a GStreamer timer to generate and put fragment metadata tags every 2 seconds
+    data->fragment_metadata_timer_id = g_timeout_add_seconds(data->put_fragment_metadata_frequency_seconds, reinterpret_cast<GSourceFunc>(put_metadata), kvssink);
```

Comment:
Q: Does the fragment length needs to be above 2 seconds when using this sample?

Reply:
Not necessarily because the first trigger would be as soon as this timer is added. This is modifiable by the customer as well